test/client.test.js

```diff
@@ -0,0 +1,221 @@
+// client.test.js
+global.WebSocket = require('ws');
+const Client = require('../client');
+const fakeData = require('./fakeData');
+
+
+describe('ClientTester', () => {
+  let client;
+  beforeEach(() => {
+      // Override _connect to return a fake ws object that doesn't try to connect.
+  Client.prototype._connect = function(url) {
+    return {
+      _url: url,
+      close: jest.fn(),
+      // You can add any other dummy methods as needed.
+    };
+  };
+    // Create a new client instance.
+    client = new Client('127.0.0.1', 17000, true);
```

Comment:
The real-time JS client takes host and port as a single url argument. Better to be consistent with that.

E.g. from the JS real-time client example:

`var client = new studio.api.Client(window.location.host, new NotificationListener());`